AirCasting/SessionsSynchronization/Controller/Interfaces/SessionUpstream.swift

```diff
@@ -0,0 +1,53 @@
+// Created by Lunar on 16/06/2021.
+//
+
+import Foundation
+import Combine
+
+/// Defines interface for objects which provide upstream connection for sessions
+///
+/// Overview of the sync process:
+/// 1. Fetch sessions stored on a device
+/// 2. Diff them against upstream database
+/// 3. Download/Upload/Delete accordingly (this interface)
+protocol SessionUpstream {
+    func upload(session: SessionsSynchronization.SessionUpstreamData) -> Future<Void, Error>
+}
+
+// MARK: Data structures
+
+extension SessionsSynchronization {
+    struct SessionUpstreamData: Equatable, Encodable {
+        typealias SensorName = String
```

Comment:
How about using this type for other models?

Reply:
Let's do it